contracts/overseer/src/contract.rs

```diff
@@ -308,16 +414,17 @@ pub fn execute_epoch_operations(deps: DepsMut, env: Env) -> Result<Response, Con
 
     // effective_deposit_rate = cur_exchange_rate / prev_exchange_rate
     // deposit_rate = (effective_deposit_rate - 1) / blocks
+    let dynrate_state_qry: DynrateState = read_dynrate_state(deps.storage)?;
     let effective_deposit_rate = epoch_state.exchange_rate / state.prev_exchange_rate;
-    let deposit_rate =
-        (effective_deposit_rate - Decimal256::one()) / Decimal256::from_uint256(blocks);
 
-    let mut messages: Vec<CosmosMsg> = vec![];
-    let mut interest_buffer = query_balance(
-        deps.as_ref(),
-        env.contract.address.clone(),
-        config.stable_denom.to_string(),
-    )?;
+    // # of blocks from the last executed height
+    let blocks = Uint256::from(env.block.height - state.last_executed_height);
+
+    let deposit_rate = update_rate(
```

Comment:
i think this should still be
```rust
    let deposit_rate =
        (effective_deposit_rate - Decimal256::one()) / Decimal256::from_uint256(blocks);
```
we're looking for the realized deposit rate, it shouldn't have anything to do with a dynamic rate calculation

Reply:
i think there might be a misunderstanding with what this function is supposed to do. currently what `execute_epoch_operations` does is:
1. perform ANC buybacks if the yield reserve increased
2. sends money from yield reserve to market contract if the realized interest rate is less than the threshold rate
3. notifies custody contracts to send staking rewards to the yield reserve

this change is supposed to add a fourth step (update the deposit rate). we need this step to be last, as we want step 2 to use the old deposit rate, and not retroactively apply the deposit rate changes. to make this more obvious, we should put `update_deposit_rate` in a separate function, that we run at the end of `execute_epoch_operations`

Reply:
done